lib/cli-engine/lint-result-cache.js

```diff
@@ -15,13 +15,31 @@ const stringify = require("json-stable-stringify-without-jsonify");
 const pkg = require("../../package.json");
 const hash = require("./hash");
 
+const debug = require("debug")("eslint:lint-result-cache");
+
 //-----------------------------------------------------------------------------
 // Helpers
 //-----------------------------------------------------------------------------
 
 const configHashCache = new WeakMap();
 const nodeVersion = process && process.version;
 
+const validCacheStrategies = ["metadata", "content"];
+const invalidCacheStrategyErrorMessage = `Cache strategy must be one of: ${validCacheStrategies
+    .map(strategy => `"${strategy}"`)
+    .join(", ")}`;
+
+/**
+ * Tests whether a provided cacheStrategy is valid
+ * @param {Object} configHelper The config helper for retrieving configuration information
+ * @returns {boolean} true if `configHelper.options.cacheStrategy` is one of `validCacheStrategies`; false otherwise
+ */
+function isValidCacheStrategy(configHelper) {
```

Comment:
Because `configHelper` is an object rather than a string, perhaps `hasValidCacheStrategy()` would be a better name for this function?

Another option would be to just pass in the cache strategy string rather than the entire configHelper object.

Reply:
You're right, I'll opt for the second option, easier to understand/reuse :)